kong/conf_loader/init.lua

```diff
@@ -1796,6 +1938,83 @@ local function load(path, custom_conf, opts)
     end
   end
 
+  -- WebAssembly module support
+  if conf.wasm then
+
+    local wasm_directives = conf["nginx_wasm_main_directives"]
+
+    local wasm_filters = get_wasm_filters(conf.wasm_filters_path)
+    conf.wasm_modules_parsed = setmetatable(wasm_filters, _nop_tostring_mt)
+
+    -- wasm vm properties are inherited from previously set directives
+    if conf.lua_ssl_trusted_certificate then
+      if #conf.lua_ssl_trusted_certificate >= 1 then
+        insert(wasm_directives, {
+          name  = "tls_trusted_certificate",
+          value = conf.lua_ssl_trusted_certificate[1],
+        })
+      end
+    end
+    if conf.lua_ssl_verify_depth and conf.lua_ssl_verify_depth > 0 then
+      insert(wasm_directives, {
+        name  = "tls_verify_cert",
+        value = "on",
+      })
+      insert(wasm_directives, {
+        name  = "tls_verify_host",
+        value = "on",
+      })
+      insert(wasm_directives, {
+        name  = "tls_no_verify_warn",
+        value = "on",
+      })
+    end
+
+    local found_proxy_wasm_lua_resolver = false
+
+    for _, directive in ipairs(conf["nginx_http_directives"]) do
+      if directive.name == "proxy_connect_timeout" then
+        insert(wasm_directives, {
+          name  = "socket_connect_timeout",
+          value = directive.value,
+        })
+      elseif directive.name == "proxy_read_timeout" then
+        insert(wasm_directives, {
+          name  = "socket_read_timeout",
+          value = directive.value,
+        })
+      elseif directive.name == "proxy_send_timeout" then
+        insert(wasm_directives, {
+          name  = "socket_send_timeout",
+          value = directive.value,
+        })
+      elseif directive.name == "proxy_buffer_size" then
+        insert(wasm_directives, {
+          name  = "socket_buffer_size",
+          value = directive.value,
+        })
+      elseif directive.name == "large_client_header_buffers" then
+        insert(wasm_directives, {
+          name  = "socket_large_buffers",
+          value = directive.value,
+        })
+      elseif directive.name == "proxy_wasm_lua_resolver" then
+        found_proxy_wasm_lua_resolver = true
+      end
+    end
```

Comment:
cc @gszr (sorry, this is something I should have caught on my review of your conf loader changes)

Ah, took me a couple glances at the surrounding code to understand, I agree with this. The role of wasm filters in the gateway is more comparable to lua plugins than anything else, so it would be more intuitive to use `lua_socket_*` settings for wasm rather than `proxy_*` settings.

The stumbling point I see here is that we support `lua_socket_*` directives in `http{}` (`nginx_http_*`) _or_ `server{}` (`nginx_proxy_*`), and deciding which of those to copy into `nginx_wasm_*` might be a headache. However, for each wasm `socket_*` directive that would go into `wasm{}` there is a corresponding `wasm_socket_*` directive that can go into `http{}` or `server{}`, which means maybe we can execute some logic like this:

* `nginx_http_lua_socket_connect_timeout` is copied to `nginx_http_wasm_socket_connect_timeout`
* `nginx_proxy_lua_socket_connect_timeout` is copied to `nginx_proxy_wasm_socket_connect_timeout`

I'm not sure where that leaves us with `nginx_http_wasm_*`/`nginx_proxy_wasm_*` properties specified by the user though--I assume we don't want to clobber those.

Reply:
If that's a larger change than we want to try to make today (understandable), then my vote would be to just remove the logic that copies directives from `nginx_http_proxy_*` into `wasm_*` and leave the rest for another time.

Reply:
Ditto - let's adjust this in a follow-up PR.